crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/AppealContributionService.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.laa.crime.contribution.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.contribution.builder.AppealContributionResponseBuilder;
+import uk.gov.justice.laa.crime.contribution.builder.CreateContributionRequestBuilder;
+import uk.gov.justice.laa.crime.contribution.builder.GetContributionAmountRequestBuilder;
+import uk.gov.justice.laa.crime.contribution.model.*;
+import uk.gov.justice.laa.crime.contribution.staticdata.enums.AssessmentResult;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AppealContributionService {
+
+    private final MaatCourtDataService maatCourtDataService;
+    private final GetContributionAmountRequestBuilder getContributionAmountRequestBuilder;
+    private final CreateContributionRequestBuilder createContributionRequestBuilder;
+    private final AppealContributionResponseBuilder appealContributionResponseBuilder;
+
+    public AppealContributionResponse calculateContribution(AppealContributionRequest appealContributionRequest, String laaTransactionId) {
+        AssessmentResult assessmentResult = determineAssessmentResult(appealContributionRequest.getAssessments());
+
+        GetContributionAmountRequest getContributionAmountRequest = getContributionAmountRequestBuilder.build(appealContributionRequest, assessmentResult);
+        BigDecimal appealContributionAmount = maatCourtDataService.getContributionAppealAmount(getContributionAmountRequest, laaTransactionId);
+
+        Integer repId = appealContributionRequest.getRepId();
+        Contribution currContribution =  maatCourtDataService.findContribution(repId, laaTransactionId);
+
+        if (currContribution.getUpfrontContributions() != appealContributionAmount) {
```

Comment:
Here you should make an `equals` comparison as the things you are comparing are objects (BigDecimal) and not primitives (or enum), see below for details (it's one of the quirks of Java)

`!currContribution.getUpfrontContributions().equals(appealContributionAmount)`

https://www.baeldung.com/java-equals-method-operator-difference